pip/utils/encoding.py

```diff
@@ -20,4 +23,9 @@ def auto_decode(data):
     for bom, encoding in BOMS:
         if data.startswith(bom):
             return data[len(bom):].decode(encoding)
+    # Lets check the first two lines as in PEP263
+    for line in data.splitlines()[:2]:
+        if line.startswith('#') and ENCODING_RE.search(line):
```

Comment:
Isn't ```data``` a bytes object here? From what I can see, ```splitlines``` behaves extremely weirdly on Python 3 when used on a bytes object.

```
>>> u'a\nb£'.encode('utf-8').splitlines()
['a', 'b\xc2\xa3']
```

You should be using bytes-specific methods here and not mixing text and bytes processing...